lib/simctl.js

```diff
@@ -326,29 +331,44 @@ async function shutdown (udid) {
   }
 }
 
+/**
+ * @typedef {Object} SimCreationOpts
+ * @property {string} platform [iOS] - Platform name in order to specify runtime such as 'iOS', 'tvOS', 'watchOS'
+ * @property {number} timeout [10000] - The maximum number of milliseconds to wait
+ *                                      unit device creation is completed.
+ */
 /**
  * Create Simulator device with given name for the particular
  * platform type and version.
  *
  * @param {string} name - The device name to be created.
  * @param {string} deviceTypeId - Device type, for example 'iPhone 6'.
  * @param {string} runtimeId - Platform version, for example '10.3'.
- * @param {number} timeout [10000] - The maximum number of milliseconds to wait
- *                                   unit device creation is completed.
+ * @param {?SimCreationOpts} opts - Simulator options for creating devives
  * @return {string} The UDID of the newly created device.
  * @throws {Error} If the corresponding simctl subcommand command
  *                 returns non-zero return code.
  */
-async function createDevice (name, deviceTypeId, runtimeId, timeout = 10000) {
+async function createDevice (name, deviceTypeId, runtimeId, opts = {}) {
+  const {
+    platform = SIM_RUNTIME_NAME_SUFFIX_IOS,
+    timeout = DEFAULT_CREATE_SIMULATOR_TIMEOUT
+  } = opts;
+
   let udid;
   // first make sure that the runtime id is the right one
   // in some versions of Xcode it will be a patch version
   try {
-    runtimeId = await getRuntimeForPlatformVersion(runtimeId);
+    runtimeId = await getRuntimeForPlatformVersion(runtimeId, platform);
   } catch (err) {
     log.warn(`Unable to find runtime for iOS '${runtimeId}'. Continuing`);
   }
 
+  const cltVersion = await getCommandLineToolsVersion();
```

Comment:
should this piece of code be located in `getRuntimeForPlatformVersion` instead?

Reply:
No. I tested the `xcrun simctl list runtimes` with differnet xcode version setting with xcode-select. Below is the result. They are different so we should not include the command line tool version check in `getRuntimeForPlatformVersion` . `getRuntimeForPlatformVersion` does not depends on Xcode version while `getRuntimeForPlatformVersion` depends.

```
kazu$ xcrun simctl list runtimes
== Runtimes ==
iOS 10.3 (10.3.1 - 14E8301) - com.apple.CoreSimulator.SimRuntime.iOS-10-3
iOS 11.4 (11.4 - 15F79) - com.apple.CoreSimulator.SimRuntime.iOS-11-4
iOS 12.0 (12.0 - 16A366) - com.apple.CoreSimulator.SimRuntime.iOS-12-0
iOS 12.2 (12.2 - 16E226) - com.apple.CoreSimulator.SimRuntime.iOS-12-2
tvOS 12.2 (12.2 - 16L225) - com.apple.CoreSimulator.SimRuntime.tvOS-12-2
watchOS 5.2 (5.2 - 16T224) - com.apple.CoreSimulator.SimRuntime.watchOS-5-2

kazu$ xcode-select -p
/Applications/Xcode.app/Contents/Developer
kazu$ sudo xcode-select -s /Applications/Xcode_10.1.app/Contents/Developer
kazu$ xcrun simctl list runtimes
== Runtimes ==
iOS 10.3 (10.3.1 - 14E8301) - com.apple.CoreSimulator.SimRuntime.iOS-10-3
iOS 11.4 (11.4 - 15F79) - com.apple.CoreSimulator.SimRuntime.iOS-11-4
iOS 12.0 (12.0 - 16A366) - com.apple.CoreSimulator.SimRuntime.iOS-12-0
iOS 12.1 (12.1 - 16B91) - com.apple.CoreSimulator.SimRuntime.iOS-12-1
tvOS 12.1 (12.1 - 16J602) - com.apple.CoreSimulator.SimRuntime.tvOS-12-1
watchOS 5.1 (5.1 - 16R591) - com.apple.CoreSimulator.SimRuntime.watchOS-5-1
```